setup-env

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+USAGE=$(cat << 'END_OF_LINE'
+This script is used to configure a developement environment for this repo.
+
+It does the following:
+  - Verifies pyenv and pyenv-virtualenv are installed.
+  - Creates a Python virtual environment.
+  - Configures the activation of the virtual enviroment for the repo directory.
+  - Installs the requirements required for development.
+  - Installs git pre-commit hooks.
+  - Configures git upstream remote "lineage" repositories.
+
+usage: setup-env [--force] [--help] [virt_env_name]
+
+END_OF_LINE
+)
+
+# Flag to force deletion and creation of virtual environment
+FORCE=0
+
+# Positional parameters
+PARAMS=""
+
+# Parse command line arguments
+while (( "$#" )); do
+  case "$1" in
+    -f|--force)
+      FORCE=1
+      shift
+      ;;
+    -h|--help)
+      echo "${USAGE}"
+      exit 0
+      ;;
+    -*) # unsupported flags
+       echo "Error: Unsupported flag $1" >&2
+       exit 1
+       ;;
+    *) # preserve positional arguments
+       PARAMS="$PARAMS $1"
+       shift
+       ;;
+      esac
+done
+
+# set positional arguments in their proper place
+eval set -- "$PARAMS"
+
+# Check to see if pyenv is installed
+if [ -z "$(which pyenv)" ] || [ -z "$(which pyenv-virtualenv)" ]; then
```

Comment:
I think it is preferred to use `command -v` vs `which` when doing this kind of check.

Reply:
I don't have strong feelings about this, since `which` is always around and its output rarely differs from that of `command -v`.  In my experience most folks use `which`, and historically that's what we have done as well.

I'm fine either way on this one.

Reply:
I've seen solid arguments against `which` such as those [here](https://unix.stackexchange.com/questions/85249/why-not-use-which-what-to-use-then) and [here](https://stackoverflow.com/questions/592620/how-can-i-check-if-a-program-exists-from-a-bash-script), so I use `command -v` when doing similar in my own scripts.

Just thought it worth mentioning, but it's not a hill I'm dying on or anything.

Reply:
Hmmm...I am feeling persuaded by [this response](https://unix.stackexchange.com/a/85250), particularly this:

> Now, on the standard front, POSIX specifies the command -v and -V commands (which used to be optional until POSIX.2008). UNIX specifies the type command (no option). That's all (where, which, whence are not specified in any standard).

and

> In the shells other than tcsh and zsh, which may tell you the path of the given executable as long as there's no alias or function by that same name in any of our ~/.cshrc, ~/.bashrc or any shell startup file and you don't define $PATH in your ~/.cshrc. If you have an alias or function defined for it, it may or may not tell you about it, or tell you the wrong thing.

Thanks @mcdonnnj!  I learned something new today.

Reply:
I too learned something today.  Thanks @mcdonnnj  !
See: 482a370